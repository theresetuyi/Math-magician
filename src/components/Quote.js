import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const Quote = ({ category }) => {
  const [quote, setQuote] = useState('');
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(false);

  useEffect(() => {
    const fetchQuote = async () => {
      try {
        setLoading(true);
        const url = 'https://api.api-ninjas.com/v1/quotes';
        const keys = 'b1PZcSwRaKAwryutCkkH2A==lVNi0KrhEV3io1wO';
        const response = await fetch(url, {
          headers: { 'X-Api-Key': keys },
          contentType: 'application/json',
        });
        const data = await response.json();
        if (data.length > 0) {
          setQuote(data[0].quote);
        }

        setLoading(false);
      } catch (error) {
        setError(false);
      }
    };
    fetchQuote();
  }, [category]);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>Error fetching quote</p>;
  }

  return (
    <div data-testId="quote-container">
      <p>{quote}</p>

    </div>
  );
};

Quote.propTypes = {
  category: PropTypes.string.isRequired,
};

export default Quote;
