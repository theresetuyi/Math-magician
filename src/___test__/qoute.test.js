import React from 'react';
import renderer from 'react-test-renderer';
import Quote from '../components/Quote';

describe('Creating Snapshot for Quote', () => {
  it('Quote should renders correctly', () => {
    const tree = renderer
      .create(<Quote category="computers" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
