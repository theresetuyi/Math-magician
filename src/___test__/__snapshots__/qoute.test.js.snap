// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Creating Snapshot for Quote Quote should renders correctly 1`] = `
<div
  data-testId="quote-container"
>
  <p />
</div>
`;
